### PR TITLE
Adapt parametrizations and losses to new API

### PR DESCRIPTION
**Pull request by saleml**
All tests pass, except those related to parametrization/loss of the Box environment (we need the fancy Distribution wrapper for that)

### CODE REVIEW

File: src/gfn/actions.py [hunk 1]
```diff
@@ -34,6 +34,10 @@ def __init__(self, tensor: TT["batch_shape", "action_shape"]):
             tensor: tensor of actions
         """
         self.tensor = tensor
+        assert len(tensor.shape) > len(self.action_shape), (
```
Comment (josephdviviano) [inline src/gfn/actions.py:37]: I know it's pedantic but I'd add the comment

`# Ensure the tensor has all action dimensions and at least one batch dimension.`

File: src/gfn/casting.py [hunk 2]
```diff
@@ -1,29 +0,0 @@
-# TODO: remove this file eventually
```
Comment (josephdviviano) [inline src/gfn/casting.py:1]: you love to see it

File: src/gfn/examples/box_utils.py [hunk 3]
```diff
@@ -377,6 +377,19 @@ def forward(
...
+    """A module to be used to create a uniform PB distribution for the Box environment
+
+    A module that returns (1, 1, 1) for all states. Used with QuarterCircle, it leads to a
+    uniform distribution over parents in the south-western part of circle.
+    """
+
+    def forward(
+        self, preprocessed_states: TT["batch_shape", 2, float]
+    ) -> TT["batch_shape", 3]:
+        return torch.ones(preprocessed_states.shape[:-1] + (3,))
```
Comment (josephdviviano) [inline src/gfn/examples/box_utils.py:390]: Worth a comment explaining the shape here (I know you also have it elsewhere, but I'm being annoying).

**Review by josephdviviano — APPROVED** [review]
In the interest of time I didn't carefully review the tests but the changes to the main code are really nice. Seems like you managed to mostly make the code shrink, which is a huge plus.
